spec-tests/src/test/java/org/candlepin/spec/consumers/ConsumerResourceActivationKeySpecTest.java

```diff
@@ -91,6 +100,131 @@ public void shouldAllowConsumerToRegisterWithActivationKeys() {
             .returns(3L, PoolDTO::getConsumed);
     }
 
+    @Test
+    @SuppressWarnings("checkstyle:methodlength")
+    public void shouldAllowConsumerToRegisterThenUnregisterWithActKeysConcurrently()
+        throws Exception {
+        // The order of keys is important. The pool ids of the first key used should be
+        // sortable by id later than those in the second key in the combined list of pools
+        OwnerDTO owner = ownerClient.createOwner(Owners.random());
+
+        int consumerCount = 3;
+        int poolCount = 6;
+        List<ConsumerDTO> consumers = Collections.synchronizedList(new ArrayList<>());
+
+        // create 6 pools and 2 activation keys
+        ActivationKeyDTO key1 = ownerClient.createActivationKey(owner.getKey(),
+            ActivationKeys.random(owner));
+        ActivationKeyDTO key2 = ownerClient.createActivationKey(owner.getKey(),
+            ActivationKeys.random(owner));
+
+        IntStream.range(0, poolCount).forEach(entry -> {
+            ProductDTO prod = Products.random()
+                .addAttributesItem(ProductAttributes.MultiEntitlement.withValue("yes"));
+            prod = ownerProductApi.createProductByOwner(owner.getKey(), prod);
+            PoolDTO pool = ownerClient.createPool(owner.getKey(), Pools.random(prod).quantity(100L));
+            if (entry < 3) {
+                activationKeyApi.addPoolToKey(key1.getId(), pool.getId(), 1L);
+            }
+            else {
+                activationKeyApi.addPoolToKey(key2.getId(), pool.getId(), 1L);
+            }
+        });
```

Comment:
Maybe we could move this to private helper method and specify the ActivationKeyDTO that should be added to pool and the number of products and pools that should be created. I'm just trying to break up this large test a bit so it is more digestible.  I know that is not easy to do given the nature of what this test is testing!

I'm thinking the method could be something like:
```
    private void createProductsAndPoolsForActivationKey(ActivationKeyDTO key, int numberOfProductAndPools) {
        IntStream.range(0, numberOfProductAndPools).forEach(entry -> {
            ProductDTO prod = Products.random()
                .addAttributesItem(ProductAttributes.MultiEntitlement.withValue("yes"));
            prod = ownerClient.createProductByOwner(owner.getKey(), prod);
            PoolDTO pool = ownerClient.createPool(owner.getKey(), Pools.random(prod).quantity(100L));
            activationKeyApi.addPoolToKey(key.getId(), pool.getId(), 1L);
        });
    }
```